lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -228,13 +247,32 @@ withNetworkLayer tr net np project k = k NetworkLayer
     currentNodeEra = handleBlockfrostError $ do
         BF.EpochInfo {_epochInfoEpoch} <- liftBlockfrost BF.getLatestEpoch
         epoch <- fromBlockfrostM _epochInfoEpoch
-        liftEither $ eraByEpoch net epoch
+        liftEither $ eraByEpoch networkId epoch
 
     timeInterpreterFromStartTime ::
         StartTime -> TimeInterpreter (ExceptT PastHorizonException IO)
     timeInterpreterFromStartTime startTime =
         mkTimeInterpreter (MsgTimeInterpreterLog >$< tr) startTime $
-            pure $ HF.mkInterpreter $ networkSummary net
+            pure $ HF.mkInterpreter $ networkSummary networkId
+
+    fetchNetworkRewardAccountBalances  ::
+        SomeNetworkDiscriminant ->
+        Set RewardAccount ->
+        IO (Map RewardAccount Coin)
+    fetchNetworkRewardAccountBalances
+        (SomeNetworkDiscriminant (Proxy :: Proxy nd)) accounts =
+        handleBlockfrostError . fmap Map.fromList $
+            for (Set.toList accounts) $ \rewardAccount -> do
+                BF.AccountInfo {..} <- liftBlockfrost $ BF.getAccount $
+                    BF.mkAddress $ encodeStakeAddress @nd rewardAccount
```

Comment:
👍  Looks much better. I hope that this is correct now, but there is little we can do besides a manual test and perhaps a second look that we do indeed have unit tests for `deodeStakeAddress` and `encodeStakeAddress`